asset_dashboard/static/js/planner.js

```diff
@@ -201,20 +238,92 @@ class PortfolioPlanner extends React.Component {
     })
   }
 
-  updatePortfolioName(e) {
-    this.setState({
-      portfolio: {...this.state.portfolio, name: e.target.value}
+  savePortfolioName(e) {
+    e.persist()
+
+    return new Promise((resolve, reject) => {
+      const nameInput = document.querySelector('#portfolio-name')
```

Comment:
in your callback, you can return the value from the form instead of getting this value from a `document.querySelector`. you should be able to access the form's value from the event: `e.target.value`.

but i just tried getting `e.target.value` from your [`updatePortfolioName` function](https://github.com/fpdcc/ccfp-asset-dashboard/blob/4ff3e767baf2dc6da2718fa1400abec6ac8eafa9/asset_dashboard/static/js/components/PortfolioTable.js#L18) and it's undefined. i think you'd need to add an `onChange` event listener to [your form's input](https://github.com/fpdcc/ccfp-asset-dashboard/blob/4ff3e767baf2dc6da2718fa1400abec6ac8eafa9/asset_dashboard/static/js/components/PortfolioTable.js#L62), get the value from `e.target.value` and set that value to the `PortfolioTable`'s state. then you can grab that value from state in your `updatePortfolioName` function and send it to the `savePortfolioName` callback. 

here's [an example from the search input](https://github.com/fpdcc/ccfp-asset-dashboard/blob/7b1c56859986abacb36be76d35d82fcfe2dcaaff/asset_dashboard/static/js/components/maps/AssetsMap.js#L129) for the asset map. it gets the value from the input change and sets it to state.

Reply:
I don't want to update the state until the name is saved, so the user can hit cancel and we don't lose the original name value. I updated this to use FormData(e.target) instead of the query selector.